cpp/src/gandiva/to_date_holder.cc

```diff
@@ -82,21 +81,15 @@ int64_t ToDateHolder::operator()(ExecutionContext* context, const std::string& d
   // Issues
   // 1. processes date that do not match the format.
   // 2. does not process time in format +08:00 (or) id.
-  struct tm result = {};
-  char* ret = strptime(data.c_str(), pattern_.c_str(), &result);
-  if (ret == nullptr) {
+  int64_t seconds_since_epoch = 0;
+  if (!internal::ParseTimestamp(data.c_str(), pattern_.c_str(), &seconds_since_epoch)) {
     return_error(context, data);
     return 0;
   }
+  // Drop any time component
+  seconds_since_epoch -= seconds_since_epoch % 86400;
```

Comment:
I'm curious: can this be negative? If so, does this give the right result?

Reply:
it should work. the mod will give a -ve result too, and so, it effectively becomes a plus.

I tried "1886-12-1  01:01:01" in to_date_holder_test, and that worked fine. I'll add the test.

Reply:
i was wrong - this is not working for -ve numbers. I'll post a fix.